geode-wan/src/main/java/org/apache/geode/cache/client/internal/locator/wan/LocatorHelper.java

```diff
@@ -46,13 +46,42 @@ public static boolean addLocator(int distributedSystemId, DistributionLocatorId
     Set<DistributionLocatorId> existingValue =
         allLocatorsInfo.putIfAbsent(distributedSystemId, locatorsSet);
     if (existingValue != null) {
-      if (!existingValue.contains(locator)) {
+      if (locator.getMemberName() != null) {
+        DistributionLocatorId tempLocator = null;
```

Comment:
This could use a comment about why the code is looking for an old ID with the same member name and removing it.  A better variable name than tempLocator would help, too.